### PR TITLE
19679 fix select2 input height

### PR DESCRIPTION
**Pull request by vcostin**
[`* `#19679` [Design] Input fields cut off at bottom (e.g. in timeline configuration)`](https://community.openproject.org/work_packages/19679)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://vcostin-bug-19679-input-fields-cut-at-bottom.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://vcostin-bug-19679-input-fields-cut-at-bottom.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/assets/stylesheets/content/_select2.scss [hunk 1]
```diff
@@ -200,3 +186,7 @@ $se2-width: 100%;
     }
   }
 }
+
+.select2-input {
+  height: rem-calc($se2-input-height) !important;
```
Comment (Azure7111) [inline app/assets/stylesheets/content/_select2.scss]: Is this a new rule/configuration not read or a result of your recent discussion @NobodysNightmare @myabc ?

Comment (floriank) [inline app/assets/stylesheets/content/_select2.scss]: I think this is normal config for `scss` - we usually use `sass`

**Comment by floriank** [comment]
can you squash the last commit @vcostin?

File: app/assets/stylesheets/content/_select2.scss (hunk 1 above)
Comment (myabc) [inline app/assets/stylesheets/content/_select2.scss]: @floriank 

> Is this a new rule/configuration not read or a result of your recent discussion

~~I think @Azure7111 was referring to our discussion about `!important` in a previous PR.~~~

Nevermind. Misread this thread.

File: app/assets/stylesheets/content/_select2.scss [hunk 2]
```diff
@@ -122,6 +116,7 @@ $se2-theme-selectable-color: #f8f8f8;
 $se2-theme-selectable-color-highlighted: $drop-down-selected-bg-color;
 
 $se2-element-height: 33px;
+$se2-input-height: 34px;
```
Comment (myabc) [inline app/assets/stylesheets/content/_select2.scss]: @vcostin you can move the `rem-calc` here.

**Comment by vcostin** [comment]
@floriank done, and lvl up, thanks :+1:

File: app/assets/stylesheets/content/_select2.scss (hunk 1 above)
Comment (myabc) [inline app/assets/stylesheets/content/_select2.scss]: `!important` is a very nasty code smell. We should avoid it wherever possible.

Comment (vcostin) [inline app/assets/stylesheets/content/_select2.scss]: @myabc I know, just had hard time to rewrite foundation defaults. Let me try one more thing to get rid of `!important`

File: app/assets/stylesheets/content/_select2.scss [hunk 3]
```diff
@@ -200,3 +195,7 @@ $se2-width: 100%;
     }
   }
 }
+
+input[type="text"].select2-input {
```
Comment (myabc) [inline app/assets/stylesheets/content/_select2.scss:199]: what was wrong with just `.select2-input`?

**Comment by floriank** [comment]
Functionality is given :+1:

File: app/assets/stylesheets/content/_select2.scss (hunk 3 above)
Comment (vcostin) [inline app/assets/stylesheets/content/_select2.scss:199]: @myabc 

```
height: 34.390625px;
input[type="text"], 
input[type="password"], 
input[type="date"],
input[type="datetime"],
input[type="datetime-local"],
input[type="month"], 
input[type="week"], 
input[type="email"], 
input[type="number"],
input[type="search"],
input[type="tel"], 
input[type="time"], 
input[type="url"], 
input[type="color"], 
textarea - 2.15rem
```

now it's fixed to 34px

Comment (floriank) [inline app/assets/stylesheets/content/_select2.scss:199]: I guess i can ignore it for the moment, but then again, my experience in CSS is sparse at best. Does this have any takedowns in the long run - and, more importantly, is there a fix for this if this was critical?

Comment (vcostin) [inline app/assets/stylesheets/content/_select2.scss:199]: This bug is reproduced only in chrome as far as I know, maybe in future versions this thing will be resolved or in foundation source.

**Pull request merged**